EIPS/eip-6785.md

```diff
@@ -0,0 +1,205 @@
+Relying on third-party platforms to obtain information regarding the utility of the NFT that one owns can lead to scams,
+phishing or other forms of fraud.
+
+Currently, utilities that are offered with NFTs, are not captured on-chain. We want the utility of an NFT to be part of
+the metadata of an NFT, and thus immutable. The metadata information would include: a) type of utility, b) description
+of utility, c) frequency and duration of utility, and d) expiration of utility. This will provide transparency as to the
+utility terms, and greater accountability on the creator to honor these utilities.
+
+As the instructions on how to access a given utility may change over time, there should be a historical record of these
+changes for transparency.
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and
+“OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Every contract compliant with [ERC-6785](./eip-6785.md) MUST implement the interface defined as follows:
+
+### Contract Interface
+
+```solidity
+// @title NFT Utility description
+///  Note: the EIP-165 identifier for this interface is 0xbcc08bd0
+
+interface IERC6785 {
+
+    // Logged when the utility description URL of an NFT is changed
+    /// @notice Emitted when the utilityURL of an NFT is changed
+    /// The empty string for `utilityUri` indicates that there is no utility associated
+    event UpdateUtility(uint256 indexed tokenId, string utilityUri);
+
+    /// @notice set the new utilityUri - remember the date it was set on
+    /// @dev The empty string indicates there is no utility
+    /// Throws if `tokenId` is not valid NFT
+    /// @param utilityUri  The new utility description of the NFT
+    /// 4a048176
+    function setUtilityUri(uint256 tokenId, string utilityUri) external;
+
+    /// @notice Get the utilityUri of an NFT
+    /// @dev The empty string for `utilityUri` indicates that there is no utility associated
+    /// @param tokenId The NFT to get the user address for
+    /// @return The utility uri for this NFT
+    /// 5e470cbc
+    function utilityUriOf(uint256 tokenId) external view returns (string memory);
+
+    /// @notice Get the changes made to utilityUri
+    /// @param tokenId The NFT to get the user address for
+    /// @return The history of changes to `utilityUri` for this NFT
+    /// f96090b9
+    function utilityHistoryOf(uint256 tokenId) external view returns (string[] memory);
+}
+```
+
+All functions defined as view MAY be implemented as pure or view
+
+Function `setUtilityUri` MAY be implemented as public or external
+
+The event `UpdateUtility` MUST be emitted when the `setUtilityUri` function is called or any other time that the utility
+of the token is changed, like in batch updates. 
+
+The `supportsInterface` method MUST return true when called with `ed231d73`
```

Comment:
This interface id doesn't match the comment on line 45. Is that intentional?

Reply:
the comment was unchanged from the previous value, before modifying the interface